ui/app/components/clients/current.js

```diff
@@ -11,9 +12,20 @@ export default class Current extends Component {
   });
   @tracked selectedNamespace = null;
 
-  // TODO CMB get from model
   get upgradeDate() {
-    return this.args.upgradeDate || null;
+    let keyInfoObject = this.args.model.versionHistory.key_info;
+    if (!keyInfoObject) {
+      return false;
+    }
+    let firstKey = Object.keys(keyInfoObject); // key name here will change, but we want the first one as this indicates when they upgraded to 1.9 or above
```

Comment:
same thing regarding naming here, and since we're calling it the "first" I think grabbing the zero index here would make sense 
`let earliestUpgradeVersion = Object.keys(keyInfoObject)[0]`

Reply:
good call.